services/rfq/api/rest/server.go

```diff
@@ -547,20 +548,33 @@ func (r *QuoterAPIServer) PutRFQRequest(c *gin.Context) {
 		span.SetAttributes(attribute.String("passive_quote_dest_amount", *passiveQuote.DestAmount))
 	}
 	quote := getBestQuote(activeQuote, passiveQuote)
+	quoteType := quoteTypeActive
+	if activeQuote == nil {
+		quoteType = quoteTypePassive
+	}
+
+	// build and return the response
+	resp := getQuoteResponse(ctx, quote, quoteType)
+	c.JSON(http.StatusOK, resp)
+}
+
+func getQuoteResponse(ctx context.Context, quote *model.QuoteData, quoteType string) (resp model.PutRFQResponse) {
+	span := trace.SpanFromContext(ctx)
 
-	// construct the response
-	var resp model.PutRFQResponse
-	if quote == nil {
+	destAmount := big.NewInt(0)
+	if quote != nil && quote.DestAmount != nil {
+		amt, ok := destAmount.SetString(*quote.DestAmount, 10)
+		if ok {
+			destAmount = amt
+		}
+	}
+	if quote == nil || destAmount.Cmp(big.NewInt(0)) <= 0 {
```

Comment:
Two small nits here:
1. `quote == nil` is redundant here, as it this condition is caught by `destAmount` being zero.
2. Why don't we use https://pkg.go.dev/math/big#Int.Sign for comparisons to zero?